src/ansi-c/parser.y

```diff
@@ -3297,8 +3297,55 @@ cprover_function_contract:
         | cprover_contract_assigns
         ;
 
+logical_equivalence_expression_list:
+          logical_equivalence_expression
+        {
+          init($$, ID_expression_list);
+          mto($$, $1);
+        }
+        | logical_equivalence_expression_list ',' logical_equivalence_expression
+        {
+          $$=$1;
+          mto($$, $3);
+        }
+        ;
+
+
+assigns_target_expression:
+          logical_equivalence_expression
+        {
+          $$=$1;
+        }
+        | logical_equivalence_expression '?' '{' logical_equivalence_expression_list '}'
+        { 
+          $$=$2;
+          parser_stack($$).id(ID_conditional_assigns_target);
+          mto($$, $1);
+          mto($$, $4);
+        }
+        | logical_equivalence_expression '?' logical_equivalence_expression
```

Comment:
Is that really what you want/expect here? I would have thought a `unary_expression` would already be a superset of what the type checker will/should accepted on the right-hand side of `?`?

Reply:
Thanks, it seems to still allow what we need and more, but less than before.